pystac_monty/sources/noaa.py

```diff
@@ -0,0 +1,158 @@
+            except Exception as e:
+                print(f"Error creating event item for SID {row.get('SID', 'unknown')}: {str(e)}")
+
+
+        return event_items
+
+    def _create_event_item_from_row(self, row):
+        if pd.isna(row.get("SID")):
+            return None
+
+        geometry = None
+        bbox = None
+        if geometry is None and not pd.isna(row.get("LAT")) and not pd.isna(row.get("LON")):
+            point = Point(float(row["LON"]), float(row["LAT"]))
+            geometry = mapping(point)
+            bbox = [float(row["LON"]), float(row["LAT"]), float(row["LON"]), float(row["LAT"])]
+        item = Item(
+            id=f"{STAC_EVENT_ID_PREFIX}{row['SID']}",
+            geometry=geometry,
+            bbox=bbox,
+            datetime=datetime.strptime(row.get("ISO_TIME"), "%Y-%m-%d %H:%M:%S"),
+            properties={},
+        )
+        MontyExtension.add_to(item)
+        monty = MontyExtension.ext(item)
+        monty.episode_number = 1  # This should be dynamically determined based on existing events
+        item.set_collection(self.get_event_collection())
+        item.properties["roles"] = ["source", "event"]
+
+
+        return item
+
+    def make_hazard_event_items(self) -> List:
+        """Create hazard items based on event items"""
+        hazard_items = []
+        event_items = self.make_source_event_items()
+
+        for event_item in event_items:
+            hazard_item = self._create_hazard_item_from_event(event_item)
+            if hazard_item:
+                hazard_items.append(hazard_item)
+
+        return hazard_items
+    
+    def _get_row_by_sid(self, sid: str) -> Optional[pd.Series]:
+        """Get original DataFrame row by SID"""
+        df = self.data.get_data()
+        matching_rows = df[df["SID"] == sid]
+        return matching_rows.iloc[0] if not matching_rows.empty else None
+
+    def _create_hazard_item_from_event(self, event_item: Item) -> Optional[Item]:
+        """Create a hazard item from an event item"""
+        hazard_item = event_item.clone()
+        hazard_item.id = hazard_item.id.replace(STAC_EVENT_ID_PREFIX, STAC_HAZARD_ID_PREFIX)
+        hazard_item.set_collection(self.get_hazard_collection())
+        hazard_item.properties["roles"] = ["source", "hazard"]
+
+        # Add hazard detail
+        monty = MontyExtension.ext(hazard_item)
+        original_row = self._get_row_by_sid(hazard_item.id.replace(STAC_HAZARD_ID_PREFIX, ""))
+        if original_row is not None:
+            monty.hazard_detail = self._create_hazard_detail(original_row)
+
+        return hazard_item
+    
+    def _create_hazard_detail(self, row: pd.Series) -> HazardDetail:
+        """Create hazard detail from row data"""
+        # First map EM-DAT classification to UNDRR-ISC codes
+
+        return HazardDetail(
+            cluster="TC",
+            severity_value="",
+            severity_unit="",
+            estimate_type=MontyEstimateType.PRIMARY,
+        )
```

Comment:
Thsi can be improved . See create_hazard_item for gdacs for more info. Follow https://github.com/IFRCGo/monty-stac-extension/blob/main/README.md#montyhazard_detail for creating hazard detail.

Reply:
may be we need to use `mag` field for severity value.